### DISCUSSION TITLE
Anisotropic mesh refinement with Fortran

### DISCUSSION THREAD

**Discussion by tpg2114** [AMReX-Codes/amrex#2487] [Q&A] [answered]
For various reasons, I am using AMReX built and linked in 3D, but I'd like to run simulations in "2D" or "1D" where the unused dimensions have `blocking_factor` cells in that direction and no more. I can do this if I don't adapt the grid, but with adaptation I'd like to specify a refinement ratio of `{2,2,1}` or `{2,1,1}`, respectively. It looks like this is supported in the C++ side of the code, but the `FAmrCore` aborts if the refinement ratios are not all equal on a given level. 

Is there a technical or design constraint that prevents anisotropic refinement? I can see that the way to access and store the refinement ratios in Fortran would need to change because right now it is a 1D array of integers and to support different values, it would need to be either a 2D array `(ndim,nlevel)` or a derived type array-of-array setup. But I'm not sure if there is a more fundamental limitation that I am overlooking.

**Comment by WeiqunZhang** [comment, marked as answer]
There are no fundamental reasons for the limitation other than we have not tried.  If you want to give this a try and it requires changes in AMReX, please feel free to submit PRs.
